### ISSUE TITLE
clang-tidy "cppcoreguidelines-pro-bounds-pointer-arithmetic" false negative

### ISSUE THREAD

**Issue by adesitter** [llvm/llvm-project#139241]
❯ clang-tidy --version
LLVM (http://llvm.org/):
  LLVM version 20.1.2
  Optimized build.

❯ cat g.cpp 
#include <span>

auto g1(std::span<int> s, std::size_t i) {
    // pointer arithmetic **not** detected
    return *(s.data() + i);
}

auto g2(std::span<int> s, std::size_t i) {
    auto p = s.data();
    // pointer arithmetic detected
    return *(p + i);
}

❯ clang-tidy "--checks=-*,cppcoreguidelines-pro-bounds-pointer-arithmetic" -quiet g.cpp -- -std=c++20 --gcc-toolchain=/usr/local/gcc-15
4 warnings generated.
.../g.cpp:11:16: warning: do not use pointer arithmetic [cppcoreguidelines-pro-bounds-pointer-arithmetic]
   11 |     return *(p + i);
      |                ^

**Comment by adesitter** [comment]
For comparison:
```
❯ clang++ -std=c++20 -c g.cpp --gcc-toolchain=/usr/local/gcc-15 -Wunsafe-buffer-usage
```
```
g.cpp:5:16: warning: unsafe pointer arithmetic [-Wunsafe-buffer-usage]
    5 |     return *(s.data() + i);
      |              ~~^~~~~~
g.cpp:5:16: note: pass -fsafe-buffer-usage-suggestions to receive code hardening suggestions
g.cpp:11:14: warning: unsafe pointer arithmetic [-Wunsafe-buffer-usage]
   11 |     return *(p + i);
      |              ^
g.cpp:11:14: note: pass -fsafe-buffer-usage-suggestions to receive code hardening suggestions
2 warnings generated.
```
